packages/intent-filler/src/strategies/basic.ts

```diff
@@ -0,0 +1,150 @@
+	 * @returns The expected profit in DAI (BigInt) which is the order fees minus the total cost
+	 */
+	async calculateProfitability(order: Order): Promise<bigint> {
+		try {
+			const { fillGas, postGas } = await this.contractService.estimateGasFillPost(order)
+			const nativeTokenPriceUsd = await this.contractService.getNativeTokenPriceUsd(order)
+
+			// 2% on top of postGas
+			const relayerFeeEth = postGas + (postGas * BigInt(200)) / BigInt(10000)
+
+			const protocolFeeUSD = await this.contractService.getProtocolFeeUSD(order, relayerFeeEth)
+
+			// fillGas and relayerFeeEth are in wei (10^18)
+			// nativeTokenPriceUsd has 18 decimals
+			const totalGasWei = fillGas + relayerFeeEth
+
+			// Converting gas cost from wei to USD using the formula:
+			// gasCostUsd = (gasWei * priceUsd) / 10^18
+			// Result has 18 decimals (matching nativeTokenPriceUsd)
+			const gasCostUsd = (totalGasWei * nativeTokenPriceUsd) / BigInt(10 ** 18)
+
+			// Both gasCostUsd and protocolFeeUSD have 18 decimals
+			const totalCostUsd = gasCostUsd + protocolFeeUSD
+
+			// Return profitability if positive, otherwise negative
+			// order.fees also has 18 decimals
+			return order.fees > totalCostUsd ? order.fees - totalCostUsd : BigInt(-1)
+		} catch (error) {
+			console.error(`Error calculating profitability:`, error)
+			return BigInt(0)
+		}
+	}
+
+	/**
+	 * Executes the order fill
+	 * @param order The order to fill
+	 * @returns The execution result
+	 */
+	async executeOrder(order: Order): Promise<ExecutionResult> {
+		const startTime = Date.now()
+
+		try {
+			const { destClient, walletClient } = this.clientManager.getClientsForOrder(order)
+
+			const postGasEstimate = await this.contractService.estimateGasForPost(order)
+			const fillOptions: FillOptions = {
+				relayerFee: postGasEstimate + (postGasEstimate * BigInt(200)) / BigInt(10000),
+			}
+
+			const ethValue = this.contractService.calculateRequiredEthValue(order.outputs)
+
+			await this.contractService.approveTokensIfNeeded(order)
+
+			const { request } = await destClient.simulateContract({
+				abi: INTENT_GATEWAY_ABI,
+				address: this.configService.getIntentGatewayAddress(order.destChain),
+				functionName: "fillOrder",
+				args: [this.contractService.transformOrderForContract(order), fillOptions as any],
+				account: privateKeyToAccount(this.privateKey),
+				value: ethValue,
+			})
+
+			const tx = await walletClient.writeContract(request)
+
+			// Wait for 10 seconds
+			await new Promise((resolve) => setTimeout(resolve, 10000))
+			const receipt = await destClient.getTransactionReceipt({ hash: tx })
+
+			const endTime = Date.now()
+			const processingTimeMs = endTime - startTime
+
+			return {
+				success: true,
+				txHash: receipt.transactionHash,
+				gasUsed: receipt.gasUsed.toString(),
+				gasPrice: receipt.effectiveGasPrice.toString(),
+				confirmedAtBlock: Number(receipt.blockNumber),
+				confirmedAt: new Date(),
+				strategyUsed: this.name,
+				processingTimeMs,
```

Comment:
On success, we should extract the post request event from the logs, store the commitment in a local SQLite DB and track its delivery with the hyperbridge SDK (In a second PR)

Reply:
Sounds good to me. 👌